### PR TITLE
endpoint_for given a URL (subdomain support broken)

### PR DESCRIPTION
**Pull request by jace**
DO NOT MERGE! Subdomain support is broken at this time.

### CODE REVIEW

File: tests/test_views.py [hunk 1]
```diff
@@ -171,3 +171,53 @@ def test_requires_permission(self):
...
+            assert endpoint_for('http://localhost/slashed/') is None
+            assert endpoint_for('http://localhost/slashed') is None
+            assert endpoint_for('http://localhost/slashed', follow_redirects=False) is None
+            assert endpoint_for('http://localhost/sub') is None
+
+            assert endpoint_for('http://example.com/') == 'index'
+            assert endpoint_for('http://example.com/slashed/') == 'slashed'
+            assert endpoint_for('http://example.com/slashed') == 'slashed'
+            assert endpoint_for('http://example.com/slashed', follow_redirects=False) is None
+            assert endpoint_for('http://example.com/sub') == 'subdomained'
```
Comment (iambibhas) [inline tests/test_views.py]: Ok. So this makes it easier to get name of the view for a url like `endpoint_for(request.url)`. We need the subdomain support for scenarios like this - given that current `SERVER_NAME` is `hasjob.co`, whether `cto.hasjob.co/slashed` is a valid url or not, and get the view name if it is.

Comment (jace) [reply tests/test_views.py]: Fixed now.

File: coaster/views/misc.py [hunk 2]
```diff
@@ -108,3 +111,49 @@ def jsonp(*args, **kw):
...
+    if not parsed_url.netloc:
+        # We require an absolute URL
+        return
+
+    # Take the current runtime environment...
+    environ = dict(request.environ)
+    # ...but replace the HTTP host with the URL's host...
+    environ['HTTP_HOST'] = parsed_url.netloc
+    # ...and the path with the URL's path (after discounting the app path, if not hosted at root).
+    environ['PATH_INFO'] = parsed_url.path[len(environ['SCRIPT_NAME']):]
```
Comment (iambibhas) [inline coaster/views/misc.py]: Do these need to be changed back to original value after this function is done with the forged request? Can this cause problem if any other piece of code called after calling `endpoint_for()` requires these environment variables?

Comment (jace) [reply coaster/views/misc.py]: No, since we made a copy with `environ = dict(request.environ)`.

Comment (iambibhas) [reply coaster/views/misc.py]: Ah, right.

**Pull request merged**